kernelci/cli/docker.py

```diff
@@ -0,0 +1,203 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2022, 2023 Collabora Limited
+# Author: Guillaume Tucker <guillaume.tucker@collabora.com>
+# Author: Jeny Sadadia <jeny.sadadia@collabora.com>
+
+"""Tool to generate Dockerfiles from Jinja2 templates and build images"""
+
+
+import io
+import json
+import sys
+
+from jinja2 import Environment, FileSystemLoader
+import docker
+
+from .base import Command, sub_main
+
+
+class DockerCommand(Command):
+    """Base command class for docker sub-commands"""
+
+    args = Command.args + [
+        {
+            'name': 'image',
+            'help': "Docker image name, e.g. gcc-10:x86",
+        },
+    ]
+    opt_args = Command.opt_args + [
+        {
+            'name': '--arch',
+            'help': "CPU architecture, e.g. x86",
+        },
+        {
+            'name': '--fragment',
+            'action': 'append',
+            'help': "Extra fragments, e.g. kernelci",
+        },
+        {
+            'name': '--image-version',
+            'help': "Docker image version tag e.g. 20221011.0",
+        },
+        {
+            'name': '--prefix',
+            'default': 'kernelci/',
+            'help': "Docker image tag prefix",
+        },
+    ]
+
+    @classmethod
+    def _gen_image_name(cls, args):
+        base_name = args.prefix + args.image
+        tag_strings = ([args.arch] if args.arch else []) + (
+            args.fragment or [])
```

Comment:
I know it was written this way in `kci_docker` but now I think the `tag_strings` should be rewritten this way for clarity as it can't really be all defined on one line with 3 conditionals:
```python
        tag_strings = [args.arch] if args.arch else []
        if args.fragment:
            tag_strings.extend(args.fragment)
        if args.image_version:
            tag_strings.append(args.image_version)
        tag_name = '-'.join(tag_strings)
```
(also `fragment` should probably become `fragments` but that's another thing).